src/figuresClasses.ts

```diff
@@ -1,19 +1,95 @@
 export interface Figure {
+  shape:string,
+  color:string,
+  getArea: () =>{}
+}
 
+function roundedDownToHundredths(area:number):number {
+  return Math.floor(area * 100) / 100;
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  public shape = 'triangle';
 
-}
+  constructor(
+    public color:string,
+    public a:number,
+    public b:number,
+    public c:number,
+  ) {
+    if (!this.isValidSides()) {
+      throw new Error('incorrect sides of triangle');
+    }
+
+    if (!this.isTriangle()) {
+      throw new Error('this is not triangle');
+    }
+
+    this.color = color;
+  }
+
+  getArea = ():number => {
+    const p = (this.a + this.b + this.c) / 2;
+    const area = Math.sqrt(p * (p - this.a) * (p - this.b) * (p - this.c));
+
+    return roundedDownToHundredths(area);
+  };
+
+  private isValidSides():boolean {
+    return this.a > 0 && this.b > 0 && this.c > 0;
+  }
+
+  private isTriangle():boolean {
+    const sides = [this.a, this.b, this.c];
 
-export class Circle {
+    sides.sort((a, b) => a - b);
 
+    return sides[0] + sides[1] > sides[2];
```

Comment:
I meant something like that:
```suggestion
    if (sides[0] + sides[1] > sides[2]) {
      throw Error...
    }
```